.github/workflows/metadata.yaml

```diff
@@ -78,3 +86,32 @@ jobs:
       - name: "Derive workspace"
         id: derive_workspace
         uses: NHSDigital/uec-dos-management/.github/actions/derive-workspace@latest
+
+  set-environment:
+    name: "Set Environment "
+    runs-on: ubuntu-latest
+    outputs:
+      environment: ${{ steps.set-env.outputs.environment }}
+    steps:
+      - name: "Determine Environment"
+        id: set-env
+        run: |
+          if [[ "${{ github.event.inputs.environment }}" != "" ]]; then
+            echo "environment=${{ github.event.inputs.environment }}" >> $GITHUB_OUTPUT
+          elif [[ "${{ github.event.label.name }}" == "Test Ready" ]]; then
+            echo "environment=test" >> $GITHUB_OUTPUT
+          elif [[ "${{ github.event.label.name }}" == "Int Ready" ]]; then
+            echo "environment=int" >> $GITHUB_OUTPUT
+          else
+            # Handling pull request closed event
+            if [[ "${{ github.event_name }}" == "pull_request" && "${{ github.event.action }}" == "closed" ]]; then
+              echo "environment=test" >> $GITHUB_OUTPUT
+            # Handling branch deletion event
+            elif [[ "${{ github.event_name }}" == "delete" && "${{ github.ref }}" == "refs/heads/task/"* ]]; then
+              echo "environment=dev" >> $GITHUB_OUTPUT
+            else
+              echo "environment=dev" >> $GITHUB_OUTPUT
+            fi
+          fi
+
+          echo "The determined environment is: $(cat $GITHUB_ENV | grep environment | cut -d '=' -f2)"
```

Comment:
GITHUB_ENV needs to be updated to GITHUB_OUTPUT

Reply:
good spot - thanks